### PR TITLE
mate-panel-applet.c: replace deprecated gtk_menu_popup

### PR DESCRIPTION
**Pull request by lukefromdc**
We need this for future porting of tray icons to standalone applets as replacements dfor gtk_menu_popup emit a "popped-up" signal when the menu is showing

Based on gnome-panel commit github.com/GNOME/gnome-panel/commit/8e6e2847eca002499e862e9ae85bbc4c16b601fa

Prior efforts at this did not work because using "NULL" for the event in gtk_menu_popup_at_widget got an invalid value and caused applet context menus to stay up only so long as the right-click mouse button was held down. GNOME fixed this the `button_press_event_new ` function

### CODE REVIEW

**Comment by lukefromdc** [comment]
Second commit removes last instance in all of mate-panel of gtk_menu_popup and (in the same function), also fixes a menu(which I had not seen before) which did not follow panel menu themes set by .mate-panel-menu-bar when invoked from the menu button. For reasons of consistancy I fixed that too.

File: mate-panel/menu.c [hunk 1]
```diff
@@ -562,11 +562,18 @@ show_item_menu (GtkWidget      *item,
...
 			     gtk_window_get_screen (GTK_WINDOW (panel_widget->toplevel)));
-
-	gtk_menu_popup (GTK_MENU (menu),
-			NULL, NULL, NULL, NULL,
-			bevent->button,
-			bevent->time);
+	/* Set up theme and transparency support */
+	GtkWidget *toplevel = gtk_widget_get_toplevel (menu);
+	/* Fix any failures of compiz/other wm's to communicate with gtk for transparency */
+	GdkScreen *screen = gtk_widget_get_screen(GTK_WIDGET(toplevel));
```
Comment (raveit65) [inline mate-panel/menu.c]: Code styling. Can you please add spaces before brackets?
Some times you do that in this block and sometimes you don't :)

**Comment by lukefromdc** [comment]
Spaces fixed, and rebased on master

**Comment by raveit65** [comment]
> Spaces fixed, and rebased on master

Hmm, not really....
```
-	gtk_menu_popup (GTK_MENU (menu),
-			NULL, NULL, NULL, NULL,
-			bevent->button,
-			bevent->time);
+	/* Set up theme and transparency support */
+	GtkWidget *toplevel = gtk_widget_get_toplevel (menu);
+	/* Fix any failures of compiz/other wm's to communicate with gtk for transparency */
+	GdkScreen *screen = gtk_widget_get_screen (GTK_WIDGET (toplevel));
+	GdkVisual *visual = gdk_screen_get_rgba_visual (screen);
+	gtk_widget_set_visual(GTK_WIDGET (toplevel), visual);
+	/* Set menu and it's toplevel window to follow panel theme */
+	GtkStyleContext *context;
+	context = gtk_widget_get_style_context (GTK_WIDGET (toplevel));
+	gtk_style_context_add_class(context,"gnome-panel-menu-bar");
+	gtk_style_context_add_class(context,"mate-panel-menu-bar");
+	gtk_menu_popup_at_pointer (GTK_MENU (menu), event);
```
Any way, it can be fixed next time when someone change code there.
No need to produce an extra commit to fix it.

**Comment by lukefromdc** [comment]
Sorry about that, that code is boilerplate dating back to summer 2015, I didn't see all of those and was never aware we wanted a space before each ( character

**Pull request merged**